sql/core/src/main/scala/org/apache/spark/sql/execution/ColumnarBatchScan.scala

```diff
@@ -109,6 +109,7 @@ private[sql] trait ColumnarBatchScan extends CodegenSupport {
 
     ctx.currentVars = null
     val rowidx = ctx.freshName("rowIdx")
+    ctx.addMutableState(ctx.JAVA_INT, rowidx)

```

Comment:
Do we have to make this global? If so, it would be good to add a comment since it is not easy to understand where `$rowidx` is used.